src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -648,6 +649,125 @@ TraceId traceId
         return result;
     }
 
+    public async Task<CascadingUpdateResult?> CascadeUpdates(
+        string resourceName,
+        long documentId,
+        short documentPartitionKey,
+        DocumentInfo documentInfo,
+        NpgsqlConnection connection,
+        NpgsqlTransaction transaction,
+        TraceId traceId
+    )
+    {
+        var result = await GetPostgresExceptionRetryPipeline()
+            .ExecuteAsync(async cancellationToken =>
+            {
+                try
+                {
+                    string BuildUpdateStatement(int n)
+                    {
+                        string retStr = $"jsonb_set(";
+                        string resourceReference =
+                            $"{char.ToLowerInvariant(resourceName[0]) + resourceName[1..]}Reference";
+                        int idx = documentInfo
+                            .DocumentIdentity.DocumentIdentityElements[n]
+                            .IdentityJsonPath.Value.LastIndexOf('.');
+                        string propertyName = documentInfo
+                            .DocumentIdentity
+                            .DocumentIdentityElements[n]
+                            .IdentityJsonPath
+                            .Value[(idx + 1)..];
+                        string newValue = documentInfo
+                            .DocumentIdentity
+                            .DocumentIdentityElements[n]
+                            .IdentityValue;
+
+                        if (n == 0)
+                        {
+                            return retStr
+                                + $"edfidoc, '{{{resourceReference}, {propertyName}}}', '\"{newValue}\"'),";
+                        }
+                        else
+                        {
+                            return retStr
+                                + BuildUpdateStatement(n - 1)
+                                + $" '{{{resourceReference}, {propertyName}}}', '\"{newValue}\"'),";
+                        }
+                    }
+
+                    foreach (var id in documentInfo.DocumentIdentity.DocumentIdentityElements)
+                    {
+                        _logger.LogInformation(id.IdentityValue);
+                    }
+
+                    var updateStatement = BuildUpdateStatement(
+                            documentInfo.DocumentIdentity.DocumentIdentityElements.Length - 1
+                        )
+                        .TrimEnd(',');
+
+                    await using var command = new NpgsqlCommand(
+                        @$"update dms.document
```

Comment:
How would this work when there is a collection of references in the document, like where BellSchedule has a collection of ClassPeriods? How do you select the correct reference in the array to update?